url_shortener/app/views.py

```diff
@@ -3,9 +3,39 @@
 
 from .forms import GenerateUrlForm
 from .models import Url
+import random
+import string
 
 def index(req):
-    return render(req, 'app/index.html')
+    if req.method == 'POST':
+        form = GenerateUrlForm(req.POST)
+        if form.is_valid():
+            url = form.save(commit=False)
+            if url.original[:8] == 'https://' or url.original[:7] == 'http://':
+                pass
+            elif url.original[:4] == 'www.':
+                url.original = 'https://' + url.original
+            else:
+                url.original = 'https://www.' + url.original
+            if len(Url.objects.filter(original = url.original)) > 0:
+                result = {'result': Url.objects.get(original=url.original).short_url}
+                return render (req, 'app/newurl.html', result)
+            possible_symbols = string.ascii_lowercase + '1234567890'
+            random_string = ''.join(random.choice(possible_symbols) for i in range(5))
```

Comment:
the generator function best would be to have as a standalone function so you could reuse it